### PR TITLE
[security] Basic Audit

### PR DESCRIPTION
**Pull request by PsiACE**
I hereby agree to the terms of the CLA available at: https://datafuse.rs/policies/cla/

## Summary

- add pr-audit workflow (will fail now, but it works)
- check dependency status (auto & quick, but not accurate enough)

## Changelog

- Build/Testing/CI

## Related Issues

Related #852

## Test Plan

No test plan.

### CODE REVIEW

File: .github/workflows/pr-audit.yml [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+    paths:
+      - '**/**/Cargo.toml'
+  pull_request:
+    paths:
+      - '**/**/Cargo.toml'
+
+jobs:
+  security-audit:
+    runs-on: ubuntu-latest
+    if: "!contains(github.event.head_commit.message, 'audit skip')"
```
Comment (PsiACE) [inline .github/workflows/pr-audit.yml:14]: Is it better to change to `if: "contains(github.event.head_commit.message, 'audit')"`?

Comment (PsiACE) [reply .github/workflows/pr-audit.yml:14]: or check `github.event.comment.body` when pr is created?

**Comment by BohuTANG** [comment]
Cool.
The only 1 vulnerability is

```
Crate:         flatbuffers
Version:       0.8.4
Title:         `read_scalar` and `read_scalar_at` allow transmuting values without `unsafe` blocks
Date:          2020-04-11
ID:            RUSTSEC-2020-0009
URL:           https://rustsec.org/advisories/RUSTSEC-2020-0009
Solution:      No safe upgrade is available!
```
Pain

File: common/datavalues/Cargo.toml [hunk 2]
```diff
@@ -16,7 +16,7 @@ common-exception = {path = "../exception"}
 # Crates.io dependencies
 anyhow = "1.0.41"
 paste = "^1.0"
-ordered-float = "2.0"
+ordered-float = "2.5"
```
Comment (BohuTANG) [inline common/datavalues/Cargo.toml:19]: Hmm, why we bump ordered-float here?

File: README.md [hunk 3]
```diff
@@ -1,7 +1,7 @@
 <div align="center">
 <h1>Datafuse</h1>
 <strong>
-Modern Real-Time Data Processing & Analytics DBMS with Cloud-Native Architecture 
+Modern Real-Time Data Processing & Analytics DBMS with Cloud-Native Architecture
```
Comment (BohuTANG) [inline README.md:4]: :+1:

File: common/datavalues/Cargo.toml (hunk 2 above)
Comment (BohuTANG) [reply common/datavalues/Cargo.toml:19]: I see.
This is scanned by https://deps.rs/repo/github/datafuselabs/datafuse
It seems that's the dep bot job, don't know why it's not working

**Comment by BohuTANG** [comment]
> Cool.
> The only 1 vulnerability is
> 
> ```
> Crate:         flatbuffers
> Version:       0.8.4
> Title:         `read_scalar` and `read_scalar_at` allow transmuting values without `unsafe` blocks
> Date:          2020-04-11
> ID:            RUSTSEC-2020-0009
> URL:           https://rustsec.org/advisories/RUSTSEC-2020-0009
> Solution:      No safe upgrade is available!
> ```
> 
> Pain

arrow-rs upstraem:
https://deps.rs/repo/github/apache/arrow-rs#vulnerabilities

We can fix it in: https://github.com/datafuse-extras/arrow-rs
And back port to the arrow-rs upstream.

**Comment by PsiACE** [comment]
> arrow-rs upstraem:
> https://deps.rs/repo/github/apache/arrow-rs#vulnerabilities
> 
> We can fix it in: https://github.com/datafuse-extras/arrow-rs
> And back port to the arrow-rs upstream.

- https://github.com/apache/arrow-rs/pull/499 OR https://github.com/datafuse-extras/arrow-rs/pull/1
- https://github.com/RustSec/advisory-db/pull/943 OR https://github.com/RustSec/advisory-db/pull/917

**Comment by BohuTANG** [comment]
@PsiACE 

We can use https://github.com/datafuse-extras/arrow-rs for this patch now :rocket:

**Review by BohuTANG — APPROVED** [review]
Look great to me

**Pull request merged**